inc/authldap.class.php

```diff
@@ -2550,6 +2577,20 @@ static function ldapImportUserByServerId(array $params, $action, $ldap_server,
                //Get information from LDAP
                if ($user->getFromLDAP($ds, $config_ldap->fields, $user_dn, addslashes($login),
                                     ($action == self::ACTION_IMPORT))) {
+                  //Get the ID by sync field (Used to check if restoration is needed)
+                  $searched_user = new User();
+                  $user_found = false;
+                  if (!($user_found = $searched_user->getFromDBbySyncField($login))) {
```

Comment:
```suggestion
                  if (!($user_found = $searched_user->getFromDBbySyncField($DB->escape($login)))) {
```

requires declaration of `global $DB`